lib/storage.js

```diff
@@ -184,7 +185,19 @@ Storage.prototype.fetchTx = function(walletId, txProposalId, cb) {
   });
 };
 
+Storage.prototype.fetchTxByHash = function(hash, cb) {
+  var self = this;
+
+  this.db.collection(collections.TXS).findOne({
+    txid: hash,
+  }, function(err, result) {
+    if (err) return cb(err);
+    if (!result) return cb();
 
+    // no need to complete data for this call use cases.
```

Comment:
I disagree. We should not make assumptions on how this method will be used. If performance takes a hit we can think of optimizing later. Do you agree?